tests/appsec/test_traces.py

```diff
@@ -33,42 +39,48 @@ def validate_appsec_span(span):
             if "_sampling_priority_v1" not in span["metrics"]:
                 raise Exception("Metric _sampling_priority_v1 should be set on traces that are manually kept")
 
-            if span["metrics"]["_sampling_priority_v1"] != APPSEC_KEEP:
-                raise Exception(f"Trace id {span['trace_id']} , sampling priority should be {APPSEC_KEEP}")
+            MANUAL_KEEP = 2
+            if span["metrics"]["_sampling_priority_v1"] != MANUAL_KEEP:
+                raise Exception(f"Trace id {span['trace_id']} , sampling priority should be {MANUAL_KEEP}")
 
-            return True
+            return validate_appsec_span_tags(span)
 
         r = self.weblog_get("/waf/", headers={"User-Agent": "Arachni/v1"})
-        interfaces.library.add_span_validation(r, validate_appsec_span)
-
+        interfaces.library.add_span_validation(r, validate_appsec_event_span_tags)
 
-@released(golang="?", dotnet="1.29.0", java="?", nodejs="2.0.0-appsec-alpha.1", php="?", python="?", ruby="0.51.0")
-class Test_AppSecMonitoring(BaseTestCase):
+    @missing_feature(library="golang", reason="appsec span tags are only applied to spans of instrumented frameworks")
     @bug(library="dotnet", reason="_dd.appsec.enabled is meta instead of metrics")
     @bug(library="ruby", reason="_dd.appsec.enabled is missing on user spans, maybe not a bug, TBC")
-    def test_events_retain_traces(self):
-        """ AppSec store in APM traces some data when enabled. """
+    def test_custom_span_tags(self):
+        """AppSec should store in APM spans some tags when enabled."""
 
-        RUNTIME_FAMILY = ["nodejs", "ruby", "jvm", "dotnet", "go", "php", "python"]
-
-        def validate_appsec_span(span):
-            if span.get("parent_id") not in (0, None):  # do nothing if not root span
+        def validate_custom_span_tags(span):
+            if span.get("name") != "init.service":
                 return
+            return validate_appsec_span_tags(span)
+
+        interfaces.library.add_span_validation(validator=validate_custom_span_tags)
+
+
+RUNTIME_FAMILY = ["nodejs", "ruby", "jvm", "dotnet", "go", "php", "python"]
+
 
-            if "_dd.appsec.enabled" not in span["metrics"]:
-                raise Exception("Can't find _dd.appsec.enabled in span's metrics")
+def validate_appsec_span_tags(span):
```

Comment:
If my remark above is right, you can move this test logic inside `validate_custom_span_tags`